### PR TITLE
workflows: autolabel language, license, versions

### PR DESCRIPTION
**Pull request by jonchang**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Autolabel more stuff.

See https://gist.github.com/jonchang/5ff6e5c20a685a0d6bf2ac47d52db3f1 for a list of affected formulae for each language.

### CODE REVIEW

**Comment by jonchang** [comment]
Script used:

```bash
cd $(brew --repo homebrew/core)/Formula
mkdir -p ~/labeling
rg -l '(?:depends_on|uses_from_macos) "perl@?.+' | sort > ~/labeling/perl.txt
rg -l '(?:depends_on|uses_from_macos) "ruby@?.+' | sort > ~/labeling/ruby.txt
rg -l 'depends_on "(?:ghc@?|haskell-stack).+' | sort > ~/labeling/haskell.txt
rg -l 'depends_on "?(?:java|openjdk@?).+' | sort > ~/labeling/java.txt
rg -l 'depends_on "go(?:@[0-9.]+)?"' | sort > ~/labeling/go.txt
rg -l 'depends_on "lua(?:@[0-9.]+|jit)?"' | sort > ~/labeling/lua.txt
rg -l 'depends_on "node(?:@\\d+)?"' | sort > ~/labeling/node.txt
rg -l 'depends_on "ocaml"' | sort > ~/labeling/ocaml.txt
rg -l 'depends_on "php@?.+' | sort > ~/labeling/php.txt
rg -l 'depends_on "python@.+' | sort > ~/labeling/python.txt
rg -l 'depends_on "rust"' | sort > ~/labeling/rust.txt
rg -l 'system "swift", "build"' | sort > ~/labeling/swift.txt
```

File: .github/workflows/autolabel.yml [hunk 1]
```diff
@@ -36,5 +36,60 @@ jobs:
...
+                    "path": "Formula/.+",
+                    "content": "depends_on \"ocaml\""
+                }, {
+                    "label": "perl",
+                    "path": "Formula/.+",
+                    "content": "(?:depends_on|uses_from_macos) \"perl@?.+"
+                }, {
+                    "label": "php",
+                    "path": "Formula/.+",
+                    "content": "depends_on \"php@?.+"
```
[the remark this answers is not in the document]
Comment (jonchang) [reply .github/workflows/autolabel.yml:77]: I didn't see any occurrences of this in the formulae. Is `uses_from_macos "php"` allowed in brew?

[the remark this answers is not in the document]
Comment (SMillerDev) [reply .github/workflows/autolabel.yml:77]: Idk, but macOS ships with a perfectly fine (though dated) PHP that can and has been used in the past.

**Review by dawidd6 — DISMISSED** [review]

File: .github/workflows/autolabel.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (dawidd6) [reply .github/workflows/autolabel.yml:77]: Had to check it, but it's true indeed: https://github.com/Homebrew/brew/blob/a7276adc10b2693482977cb08cb8c8989fc89940/Library/Homebrew/rubocops/uses_from_macos.rb#L62

**Review by MikeMcQuaid — DISMISSED** [review]
Love this.

**Comment by jonchang** [comment]
Thanks for the reviews! On mobile so can't merge but the changes look good.

**Comment by dawidd6** [comment]
I just merged it from my phone :)

**Comment by jonchang** [comment]
I think the button disappears on the GitHub app under some conditions. Thanks for the assist!

**Comment by Rylan12** [comment]
I think there's an issue with the `missing license` label. It was incorrectly added to #58915

**Comment by dawidd6** [comment]
It was added to every PR actually, lol. Gonna delete this constraint real quick, dunno how to fix it now.

**Comment by dawidd6** [comment]
https://github.com/Homebrew/homebrew-core/commit/d52423aba6c6a2a88517ee1f551001c63fb46480

cc @jonchang

**Comment by jonchang** [comment]
I tested this with ripgrep, but thinking about it, the behavior of the dot special character in regexes probably differs from JavaScript for multiline matches, in that ripgrep excludes the newline character. Try replacing the `license .+` construct with `license \"[a-zA-Z0-9-_]+\"`. Or just remove the bit, it's not really that important.

**Comment by dawidd6** [comment]
Yea, I suggest reading a formula content in `node` and testing the regex with `.match()` there.

**Comment by jonchang** [comment]
Away from computer for the next 8 hours 😅

Just revert it and I'll take a peek tomorrow.

**Comment by Rylan12** [comment]
> Try replacing the `license .+` construct with `license \"[a-zA-Z0-9-_]+\"`

The issue, I think, is that it was matching `\n\n` (i.e. blank lines)

This would have the same issue I believe

**Pull request merged**